src/components/date/__internal__/date-picker/date-picker.component.tsx

```diff
@@ -82,6 +87,13 @@ export const DatePicker = ({
   pickerTabGuardId,
   onPickerClose,
 }: DatePickerProps) => {
+  Logger.warn(
+    "Both minDate and maxDate must be provided to DatePicker component in order to set disabled dates.",
+  );
```

Comment:
**suggestion**: for the warning message, what do you think about this tweak? By introducing a flag like `minMaxDateWarningLogged`, the warning would only appear once when the condition is met.

This could make the logging more consumer-friendly and prevent repeated warnings, especially in case the component frequently re-renders:

```tsx
let minMaxDateWarningLogged = false;

const DatePicker = () => {
  if (!!minDate !== !!maxDate) {
    Logger.warn(
      "Both minDate and maxDate must be provided to DatePicker component in order to set disabled dates."
    );
    minMaxDateWarningLogged = true;
  }

  // component code...
};
```